### PR TITLE
12244_comment question

### PR DESCRIPTION
**Pull request by tuananhp-1844**
https://edu-redmine.sun-asterisk.vn/issues/12244

### CODE REVIEW

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -36,4 +36,14 @@ public function __construct()
     {
         $this->middleware('guest')->except('logout');
     }
+
+    public function showLoginForm()
+    {
+        if(!session()->has('url.intended'))
```
Comment (framgiaciv3) [inline app/Http/Controllers/Auth/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/Auth/LoginController.php</br>**Line**: 42</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/Auth/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/Auth/LoginController.php</br>**Line**: 42</br>Expected 1 space(s) after closing parenthesis; found newline

File: app/Http/Controllers/Auth/LoginController.php [hunk 2]
```diff
@@ -36,4 +36,14 @@ public function __construct()
...
+        {
+            session(['url.intended' => url()->previous()]);
+        }
+        
+        return view('auth.login');
```
Comment (framgiaciv3) [inline app/Http/Controllers/Auth/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Auth/LoginController.php</br>**Line**: 47</br>Whitespace found at end of line

File: app/Http/Controllers/Questions/QuestionController.php [hunk 3]
```diff
@@ -15,12 +16,19 @@ class QuestionController extends Controller
...
+    private $answerRepository;
 
-    public function __construct(QuestionInterface $question, CategoryInterface $category, UserInterface $user)
+    public function __construct(
+        QuestionInterface $question,
+        CategoryInterface $category,
+        UserInterface $user,
+        AnswerInterface $answerRepository
+    )
     {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 27</br>The closing parenthesis and the opening brace of a multi-line function declaration must be on the same line

File: app/Http/Controllers/Questions/QuestionController.php [hunk 4]
```diff
@@ -86,15 +94,16 @@ public function store(CreateQuestionRequest $request)
...
         $question = $this->questionRepository->findOrFail($id);
-        $question = $question->load(['user', 'answers.user', 'votes', 'tags', 'polls.users']);
+        $question = $question->load(['user', 'votes', 'tags', 'polls.users']);
         $this->questionRepository->increaseView($id);
         $relate = $this->questionRepository->relate($question, config('pagination.question'));
         $questionClips = [];
         if (Auth::check()) {
             $questionClips = Auth::user()->clips->sortBy('id')->take(config('paginate.question'));
         }
+        $comments = $question->answers()->orderBy('id', 'DESC')->where('parent_id', 0)->with('user', 'childs')->paginate(config('pagination.comment'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 104</br>Line exceeds 120 characters; contains 151 characters

File: app/Http/Controllers/Auth/LoginController.php [hunk 5]
```diff
@@ -36,4 +36,13 @@ public function __construct()
     {
         $this->middleware('guest')->except('logout');
     }
+
+    public function showLoginForm()
+    {
+        if (!session()->has('url.intended')){
```
Comment (framgiaciv3) [inline app/Http/Controllers/Auth/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/Auth/LoginController.php</br>**Line**: 42</br>Expected 1 space(s) after closing parenthesis; found 0

File: app/Http/Controllers/Auth/LoginController.php [hunk 6]
```diff
@@ -36,4 +36,13 @@ public function __construct()
...
+            session(['url.intended' => url()->previous()]);
+        }
+        
+        return view('auth.login');
```
Comment (framgiaciv3) [inline app/Http/Controllers/Auth/LoginController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Auth/LoginController.php</br>**Line**: 46</br>Whitespace found at end of line

File: app/Http/Controllers/Questions/QuestionController.php [hunk 7]
```diff
@@ -15,12 +16,18 @@ class QuestionController extends Controller
...
     private $categoryRepository;
     private $userRepository;
+    private $answerRepository;
 
-    public function __construct(QuestionInterface $question, CategoryInterface $category, UserInterface $user)
+    public function __construct(
+        QuestionInterface $question,
+        CategoryInterface $category,
+        UserInterface $user,
+        AnswerInterface $answerRepository)
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 25</br>The closing parenthesis of a multi-line function declaration must be on a new line

File: app/Http/Controllers/Questions/QuestionController.php [hunk 8]
```diff
@@ -15,12 +16,18 @@ class QuestionController extends Controller
...
     {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 26</br>The closing parenthesis and the opening brace of a multi-line function declaration must be on the same line

File: app/Http/Controllers/Questions/QuestionController.php (hunk 3 above)
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 27</br>The closing parenthesis and the opening brace of a multi-line function declaration must be on the same line

File: resources/views/questions/show.blade.php [hunk 9]
```diff
@@ -232,27 +234,45 @@
...
     </div><!-- End page-content -->
-
+    
+    @if (Auth::check())
     <div id="respond" class="comment-respond page-content clearfix">
         <div class="boxedtitle page-title">
             <h2>{{ __('Leave a reply') }}</h2>
         </div>
-        <form action="#" method="post" id="commentform" class="comment-form">
+        <form action="{{ route('questions.answers.store', ['questions' => $question->id]) }}" method="post" id="commentform" class="comment-form">
```
Comment (tranducquoc) [inline resources/views/questions/show.blade.php]: không dùng form laravel colective hả em

File: routes/web.php [hunk 10]
```diff
@@ -46,3 +46,5 @@
 Route::resource('questions.clips', 'Questions\ClipController')->only('index');
 
 Route::resource('polls.users', 'Users\PollController')->only('store');
+
+Route::resource('questions.answers', 'Questions\AnswerController');
```
Comment (tranducquoc) [inline routes/web.php]: chỉ ra action nào mình dùng để hạn chế thừa url

**Pull request merged**